geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/export/formats/LeafletMapExporter.scala

```diff
@@ -0,0 +1,185 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.tools.export.formats
+
+import java.io._
+
+import com.typesafe.scalalogging.LazyLogging
+import com.vividsolutions.jts.geom.{Coordinate, Geometry}
+import org.geotools.geojson.feature.FeatureJSON
+import org.locationtech.geomesa.tools.Command.user
+import org.locationtech.geomesa.tools.export.formats.LeafletMapExporter.SimpleCoordinate
+import org.locationtech.geomesa.tools.export.{ExportCommand, ExportParams}
+import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.locationtech.geomesa.tools.export.formats.LeafletMapExporter._
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+import scala.io.Source
+import scala.io.StdIn.readLine
+
+class LeafletMapExporter(params: ExportParams) extends FeatureExporter with LazyLogging {
+
+  private val json = new FeatureJSON()
+  private val coordMap = mutable.Map[SimpleCoordinate[Double], Int]()
+
+  private var first = true
+  private var sft: SimpleFeatureType = _
+  private var totalCount: Long = 0L
+
+  Option(params.maxFeatures) match {
+    case Some(limit) =>
+      if (limit > 10000) {
+        // Note we have to ask if the user wants to continue like this or the output will be
+        // hidden since this command runs in a sub-shell
+        user.warn("A large number of features might be exported. This can cause performance issues when using the map. For large numbers of features it is recommended to use GeoServer to render a map.\nDo you want to continue? [y/N]")
+        // readLine() could be string, null or empty string, handle all
+        val response = Option(readLine()).getOrElse("").headOption.getOrElse("n").toString
+        if (response.matches("^[nN]")) {
+          sys.exit(1)
+        }
+      }
+    case None =>
+  }
+
+  Option(params.gzip) match {
+    case Some(_) => user.warn("GZip parameter cannot be used with leaflet format, ignoring.")
```

Comment:
minor - since you're not using the unapply, or the alternate match statements, it might be clearer to just write
`if (params.gzip != null) { logger.warn(...) }`